src/mon/AuthMonitor.cc

```diff
@@ -1106,9 +1106,6 @@ bool AuthMonitor::valid_caps(const vector<string>& caps, ostream *out)
       if (!mdscap.parse(g_ceph_context, *(p+1), out)) {
 	return false;
       }
-    } else {
-      *out << "unknown cap type '" << *p << "'";
-      return false;
```

Comment:
why should we allow unrecognized services?

Reply:
to be perfectly honest, I'm not entirely sure whether we should or should not. The prototype's comment in the header file states we are going to ignore unrecognized services though, and I see calls to `valid_caps()` that I imagine may contain unrecognized services (e.g., `auth add`, `auth caps`, etc). If we do not allow unrecognized services, then we must be very sure we're either not using caps for other services (which we clearly do), or go through each one of the commands using caps and find ways to validate other caps that are not mon, mgr, osd or mds.

Reply:
whelp, when I say "we clearly do", I should have said "I think we do"; then again, maybe we don't. Do we?

Reply:
I don't think we do.  We might let people store them, but they can't be used because the auth code has a hard coded set of services it recognizes.

IMO we should leave this new restriction in place and fix the header comment you saw?

Reply:
that works for me.